### DISCUSSION TITLE
Input previously captured static image

### DISCUSSION THREAD

**Discussion by akhalid** [journeyapps/zxing-android-embedded#671] [Q&A] [answered]
Hello,

Is there a way to input a previously captured static image (QR Code)? Say a bitmap?

Thanks!

**Comment by rkistner** [comment, marked as answer]
Technically yes - you can see some previous discussions & examples:

https://github.com/journeyapps/zxing-android-embedded/issues/254
https://github.com/journeyapps/zxing-android-embedded/issues/388

But note that:
1. You don't really need this library to do that - you can use the underlying `zxing:core` library directly.
2. The underlying algorithms aren't optimized to scan from a static image - it is designed to do many scan attempts quickly from a stream of images, instead of doing optimal scanning for a single image. So the results may not be as good as you'd want.

**Reply by akhalid** [reply]
Got it. I will look into it. Thank you!
